lib/ansible/modules/database/postgresql/postgresql_user.py

```diff
@@ -354,9 +354,7 @@ def user_alter(cursor, module, user, password, role_attr_flags, encrypted, expir
         new_role_attrs = cursor.fetchone()
 
         # Detect any differences between current_ and new_role_attrs.
-        for i in range(len(current_role_attrs)):
-            if current_role_attrs[i] != new_role_attrs[i]:
-                changed = True
+        changed = current_role_attrs != new_role_attrs
```

Comment:
`psycopg2.extras.DictRow` was already here in [1.99.12](https://github.com/psycopg/psycopg2/blob/1_99_12/lib/extras.py#L61) (which was released in march 2005).